### DISCUSSION TITLE
"Incoming flows do not join' error shown at the 'Task update application status' in the workflow process

### DISCUSSION THREAD

**Discussion by amalinamohds** [AOT-Technologies/forms-flow-ai#1802] [Q&A]
Hi, I intend to use the formsflow to create a public form that can send an email notification to the user who submits the form.

I had tried to set up a workflow process that allowed to send an email notification to users after they submitted the form. But there are some errors triggered at the 'Task update application status'  which is "Incoming flows do not join', so during testing submission, the submission form was failed. 

Is there anything that needs to be done at this part? Please help me to find a solution for this issue. Has anyone experienced this issue before or can help me how to setup the workflow process to enable the email notification to be sent to the user? Kindly refer below image as reference.

![formsflow email](https://github.com/AOT-Technologies/forms-flow-ai/assets/34720353/9588da9e-681e-4f3b-aa38-ca2f659fda75)

**Comment by Bonymol-aot** [comment]
Hi @amalinamohds ,  you're dealing with a warning related to workflow structure and the suggestion is to add an XOR gateway for handling the 'approve' and 'reject' flows as shown below. The warning itself doesn't impact the functionality of the workflow. However, it's recommended to address it to improve the workflow design.

![image](https://github.com/AOT-Technologies/forms-flow-ai/assets/88372115/61a48a69-85cf-4901-90cf-4ce5b20aef68)

Although the warning is not the cause of the submission failure, it's advised to thoroughly check the detailed logs in the forms-flow-bpm  to identify the actual error leading to the submission failure.
